### PR TITLE
#8 init_commit_of_generator_code

### PR DESCRIPTION
**Pull request by azamifarda**

### CODE REVIEW

File: asid_v2/keras_classes.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        'Generates data containing batch_size samples' # X : (n_samples, *dim)
+        # Initialization
+        X = np.empty((self.batch_size, *self.dims_input))
+        y = np.empty((self.batch_size, *self.dims_output))
+        z = np.empty((self.batch_size, *self.dims_amsr2))
+
+        # Generate data
+        for i, ID in enumerate(list_IDs_temp):
+
+            X[i,:,:,0] = np.load('/workspaces/ASID-v2-builder/output/' + ID ).get(self.input_var_name)
```
Comment (akorosov) [inline asid_v2/keras_classes.py:53]: Path should not be hardcoded.

File: asid_v2/keras_script.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+precentage_of_training = .8
+train_sublist_id_list = id_list[: int(len(id_list) * precentage_of_training)]
+valid_sublist_id_list = id_list[int(len(id_list) * precentage_of_training) :]
+print(f"total number of training samples: {len(train_sublist_id_list)}")
+print(f"total number of validation samples: {len(valid_sublist_id_list)}")
+# Datasets
+partition = {'train': train_sublist_id_list, 'validation': valid_sublist_id_list}
+input_var_name = 'nersc_sar_primary'
+output_var_name = 'CT'
+amsr2_var_name = 'btemp_6_9h'
```
Comment (akorosov) [inline asid_v2/keras_script.py:36]: Several amsr2 bands should be used

File: asid_v2/keras_script.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#input_ = layers.Input(shape=(*np.load('/workspaces/ASID-v2-builder/output/' + id_list[0] ).get('sar_primary').shape, 1))
+x = layers.BatchNormalization()(input_)
+x = layers.Conv2D(8, (3, 3), padding='same', activation='relu')(x)
+x = layers.AveragePooling2D(pool_size=(2,2), strides=(2,2))(x)
+x = layers.Conv2D(16, (3, 3), padding='same', activation='relu')(x)
+x = layers.AveragePooling2D(pool_size=(2,2), strides=(2,2))(x)
+x = layers.Conv2D(32, (3, 3), padding='same', activation='relu')(x)
+x = layers.AveragePooling2D(pool_size=(7,7), strides=(6,6))(x)
+x = layers.AveragePooling2D(pool_size=(2,2), strides=(2,2))(x)
+x = layers.UpSampling2D(size=(10, 10))(x+input_2)
```
Comment (akorosov) [inline asid_v2/keras_script.py:66]: input_2 should be stacked to x in this example in the z direction

from 32 x 13 x 13 and N x 13 x 13 into 32+N x 13 x 13

**Review by akorosov — CHANGES_REQUESTED** [review]
Few minor things

File: asid_v2/keras_classes.py (hunk 1 above)
Comment (azamifarda) [reply asid_v2/keras_classes.py:53]: Done

File: asid_v2/keras_script.py (hunk 2 above)
Comment (azamifarda) [reply asid_v2/keras_script.py]: Done

File: asid_v2/keras_script.py (hunk 3 above)
Comment (azamifarda) [reply asid_v2/keras_script.py]: Done

File: asid_v2/keras_script.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        id_list.append(x)
+random.shuffle(id_list)
+precentage_of_training = .8
+train_sublist_id_list = id_list[: int(len(id_list) * precentage_of_training)]
+valid_sublist_id_list = id_list[int(len(id_list) * precentage_of_training) :]
+print(f"total number of training samples: {len(train_sublist_id_list)}")
+print(f"total number of validation samples: {len(valid_sublist_id_list)}")
+# Datasets
+partition = {'train': train_sublist_id_list, 'validation': valid_sublist_id_list}
+input_var_name = 'nersc_sar_primary'
```
Comment (akorosov) [inline asid_v2/keras_script.py:32]:
```suggestion
input_var_names = ['nersc_sar_primary', 'nersc_sar_secondary']
```

**Review by akorosov — CHANGES_REQUESTED** [review]
Sorry, I forgot - SAR may also have two bands, primary and secondary. Can you please change the DataGenerator accordingly?

File: asid_v2/keras_script.py (hunk 4 above)
Comment (azamifarda) [reply asid_v2/keras_script.py]: Done

**Review by akorosov — APPROVED** [review]

**Pull request merged**